crates/ruff_linter/src/rules/pyupgrade/rules/quoted_annotation.rs

```diff
@@ -87,14 +87,14 @@ pub(crate) fn quoted_annotation(checker: &mut Checker, annotation: &str, range:
     let placeholder_range = TextRange::up_to(annotation.text_len());
     let spans_multiple_lines = annotation.contains_line_break(placeholder_range);
 
-    let tokenizer = SimpleTokenizer::new(annotation, placeholder_range);
-    let last_token_is_comment = matches!(
-        tokenizer.last(),
-        Some(SimpleToken {
-            kind: SimpleTokenKind::Comment,
-            ..
-        })
-    );
+    let last_token_is_comment = checker
+        // Inside an annotation, the `checker.tokens()`
+        // returns the tokens for the parsed annotation.
+        .tokens()
```

Comment:
nit: I'd remove this comment as the information exists as docs to `checker.tokens` method